File: apps/web/src/lib/api/exchange.ts
import api from './client';

export interface ExchangeRequest {
  id: string;
  order_id: string;
  order_item_id: string;
  reason_code: string;
  reason_detail: string | null;
  desired_option: Record<string, unknown> | null;
  status: string;
  return_tracking_number: string | null;
  return_carrier: string | null;
  new_tracking_number: string | null;
  new_carrier: string | null;
  seller_response: string | null;
  responded_at: string | null;
  created_at: string;
}

export interface OrderInquiry {
  id: string;
  order_id: string;
  inquiry_type: string;
  title: string;
  content: string;
  images: string[];
  status: string;
  seller_reply: string | null;
  replied_at: string | null;
  created_at: string;
}

export const createExchange = (data: {
  order_id: string;
  order_item_id: string;
  reason_code: string;
  reason_detail?: string;
  desired_option?: Record<string, unknown>;
}) => api.post<{ data: { id: string } }>('/exchanges', data).then((r) => r.data);

export const getMyExchanges = (params?: { page?: number; per_page?: number }) =>
  api.get<{ data: ExchangeRequest[] }>('/exchanges', { params }).then((r) => r.data);

export const updateExchangeTracking = (id: string, data: { return_tracking_number: string; return_carrier: string }) =>
  api.put(`/exchanges/${id}/tracking`, data).then((r) => r.data);

export const getSellerExchanges = (params?: { page?: number; per_page?: number }) =>
  api.get<{ data: ExchangeRequest[] }>('/seller/exchanges', { params }).then((r) => r.data);

export const respondExchange = (id: string, data: {
  status: string;
  seller_response?: string;
  new_tracking_number?: string;
  new_carrier?: string;
}) => api.put(`/seller/exchanges/${id}/respond`, data).then((r) => r.data);

export const createInquiry = (data: {
  order_id: string;
  inquiry_type: string;
  title: string;
  content: string;
  images?: string[];
}) => api.post<{ data: { id: string } }>('/inquiries', data).then((r) => r.data);

export const getMyInquiries = (params?: { page?: number; per_page?: number }) =>
  api.get<{ data: OrderInquiry[] }>('/inquiries', { params }).then((r) => r.data);

export const getSellerInquiries = (params?: { page?: number; per_page?: number }) =>
  api.get<{ data: OrderInquiry[] }>('/seller/inquiries', { params }).then((r) => r.data);

export const replyInquiry = (id: string, seller_reply: string) =>
  api.put(`/seller/inquiries/${id}/reply`, { seller_reply }).then((r) => r.data);

File: packages/backend/src/domain/exchange.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, FromRow)]
pub struct ExchangeRequest {
    pub id: Uuid,
    pub order_id: Uuid,
    pub order_item_id: Uuid,
    pub buyer_id: Uuid,
    pub seller_id: Uuid,
    pub reason_code: String,
    pub reason_detail: Option<String>,
    pub desired_option: Option<serde_json::Value>,
    pub status: String,
    pub return_tracking_number: Option<String>,
    pub return_carrier: Option<String>,
    pub new_tracking_number: Option<String>,
    pub new_carrier: Option<String>,
    pub seller_response: Option<String>,
    pub responded_at: Option<DateTime<Utc>>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Deserialize)]
pub struct CreateExchangeRequest {
    pub order_id: Uuid,
    pub order_item_id: Uuid,
    pub reason_code: String,
    pub reason_detail: Option<String>,
    pub desired_option: Option<serde_json::Value>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateExchangeStatusRequest {
    pub status: String,
    pub seller_response: Option<String>,
    pub new_tracking_number: Option<String>,
    pub new_carrier: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct ExchangeTrackingRequest {
    pub return_tracking_number: String,
    pub return_carrier: String,
}

// Order Inquiry types
#[derive(Debug, Serialize, FromRow)]
pub struct OrderInquiry {
    pub id: Uuid,
    pub order_id: Uuid,
    pub buyer_id: Uuid,
    pub seller_id: Uuid,
    pub inquiry_type: String,
    pub title: String,
    pub content: String,
    pub images: Option<serde_json::Value>,
    pub status: String,
    pub seller_reply: Option<String>,
    pub replied_at: Option<DateTime<Utc>>,
    pub resolved_at: Option<DateTime<Utc>>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Deserialize)]
pub struct CreateInquiryRequest {
    pub order_id: Uuid,
    pub inquiry_type: String,
    pub title: String,
    pub content: String,
    pub images: Option<Vec<String>>,
}

#[derive(Debug, Deserialize)]
pub struct ReplyInquiryRequest {
    pub seller_reply: String,
}
